src/screencast/wlr_screencast.c

```diff
@@ -325,28 +344,36 @@ static void wlr_registry_handle_add(void *data, struct wl_registry *reg,
 		uint32_t id, const char *interface, uint32_t ver) {
 	struct xdpw_screencast_context *ctx = data;
 
+	logprint(DEBUG, "wlroots: interface to register %s  (Version: %u)",interface, ver);
 	if (!strcmp(interface, wl_output_interface.name)) {
 		struct xdpw_wlr_output *output = malloc(sizeof(*output));
 
 		output->id = id;
-		output->output = wl_registry_bind(reg, id, &wl_output_interface, 1);
+		logprint(DEBUG, "wlroots: |-- registered to interface %s (Version %u)", interface, WL_OUTPUT_VERSION);
+		output->output = wl_registry_bind(reg, id, &wl_output_interface, WL_OUTPUT_VERSION);
 
 		wl_output_add_listener(output->output, &wlr_output_listener, output);
 		wl_list_insert(&ctx->output_list, &output->link);
 	}
 
 	if (!strcmp(interface, zwlr_screencopy_manager_v1_interface.name)) {
+		uint32_t version = SC_MANAGER_VERSION < ver ? SC_MANAGER_VERSION :
+			ver < SC_MANAGER_VERSION_MIN ? SC_MANAGER_VERSION_MIN : ver;
```

Comment:
I am guilty of trying to get away with one-liners like this too, but this is way more complicated to read than:
```c
uint32_t version = ver;
if (SC_MANAGER_VERSION < ver) {
  version = SC_MANAGER_VERSION;
}
if (SC_MANAGER_VERSION_MIN > ver) {
  version = SC_MANAGER_VERSION_MIN;
}
```
Also, I think we need to bail out if the version offered is less than the minimum supported. Similarly, we haven't been checking for the other interfaces if the version offered is compatible with the version(s) we support. @emersion, are these interfaces necessarily backwards compatible?

Reply:
>Similarly, we haven't been checking for the other interfaces if the version offered is compatible with the version(s) we support.

If we request v3 and the compositor only supports v2, I think libwayland will error out. Would need to double-check.

> @emersion, are these interfaces necessarily backwards compatible?

Yes, but not in a way you'd necessarily expect.

- If the compositor supports v3, it *must* support v1 and v2 as well
- If the client requests v3, the compositor may behave differently than if the client requests v1 or v2 (not necessarily in a backwards-compatible way)

Reply:
> I am guilty of trying to get away with one-liners like this too, but this is way more complicated to read than:

Sure, changed it.

Reply:
Looks good, but given emersion's comment, might be a good idea to look into handling errors arising from `bind`ing a version that is too high (or I suppose we could just call exit() or similar if the compositor doesn't offer the minimum required version).